src/Microsoft.Security.Utilities.Core/DataClassification/DAT101_002_Ip6.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System.Collections.Generic;
+
+namespace Microsoft.Security.Utilities
+{
+    public class Ip6 : RegexPattern
+    {
+        public Ip6()
+        {
+            Id = "DAT101/002";
+            Name = nameof(Ip6);
+            Pattern = @"(?i)^([0-9a-f]{1,4}:){7}[0-9a-f]{1,4}$";
+        }
+
+        public override IEnumerable<string> GenerateTruePositiveExamples()
+        {
+            var example = "3fff:fff:ffff:ffff:ffff:ffff:ffff:ffff";
+            yield return example.ToLowerInvariant();
```

Comment:
>ToLowerInvariant [](http://example.com/codeflow?start=33&length=16)

example is already lower-case, why generate a short-lived string that duplicates the test case?